services/oauth/v2/utils.go

```diff
@@ -0,0 +1,95 @@
+package v2
+
+import (
+	"bytes"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"time"
+
+	jsoniter "github.com/json-iterator/go"
+	"github.com/samber/lo"
+	"github.com/tidwall/gjson"
+
+	router_utils "github.com/rudderlabs/rudder-server/router/utils"
+	"github.com/rudderlabs/rudder-server/utils/misc"
+)
+
+var jsonfast = jsoniter.ConfigCompatibleWithStandardLibrary
+
+const (
+	CategoryRefreshToken = "REFRESH_TOKEN"
+	// CategoryAuthStatusInactive Identifier to be sent from destination(during transformation/delivery)
+	CategoryAuthStatusInactive = "AUTH_STATUS_INACTIVE"
+	// RefTokenInvalidGrant Identifier for invalid_grant or access_denied errors(during refreshing the token)
+	RefTokenInvalidGrant = "ref_token_invalid_grant"
+)
+
+var (
+	ErrorCategories             = []string{CategoryRefreshToken, CategoryAuthStatusInactive}
```

Comment:
Is the suggestion provided to make this into an O(1) operation ?

Reply:
Yes. Probably we might be adding more things in future. Since the errors are limited, I would prefer slice over arrays because memory wise there will not be much difference but with increased number of errors, map would be better over slice.

Reply:
Don't think we would be adding more categories in future. But I will make the changes for the possibility